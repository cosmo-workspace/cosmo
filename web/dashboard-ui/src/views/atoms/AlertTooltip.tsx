import {
  darken, lighten, styled, Tooltip, tooltipClasses, TooltipProps
} from "@mui/material";
import React from "react";

export const AlertTooltip = styled(({ className, ...props }: TooltipProps) => (
  <Tooltip {...props} arrow classes={{ popper: className }} />
))(({ theme }) => ({
  [`& .${tooltipClasses.arrow}`]: {
    fontSize: '1rem',
    color: (theme.palette.mode === 'light' ? lighten : darken)(theme.palette['info'].light, 0.9),
  },
  [`& .${tooltipClasses.tooltip}`]: {
    borderRadius: "4px",
    boxShadow: theme.shadows[5],
    backgroundColor: (theme.palette.mode === 'light' ? lighten : darken)(theme.palette['info'].light, 0.9),
  },
}));
