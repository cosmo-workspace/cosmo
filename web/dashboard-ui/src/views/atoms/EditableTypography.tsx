import { Check, Close, Edit } from "@mui/icons-material";
import {
  IconButton,
  InputBase,
  InputBaseProps,
  Stack,
  Tooltip,
} from "@mui/material";
import React, { useState } from "react";

export type EditableTypographyProps = InputBaseProps & {
  children: string;
  onSave: (inputData: string) => void;
  showAlways?: boolean;
};

export const EditableTypography: React.FC<EditableTypographyProps> = ({
  children,
  onSave,
  showAlways,
  ...props
}) => {
  const [showEditIcon, setShowEditIcon] = useState(showAlways);
  const [editting, setEditing] = useState(false);
  const [inputData, setInputData] = useState<string>(children);

  const editIcon = (
    <Tooltip title="Edit" placement="top">
      <IconButton
        disableRipple
        size="small"
        type="button"
        sx={{ ml: 1 }}
        onClick={() => {
          setEditing(true);
        }}
      >
        <Edit />
      </IconButton>
    </Tooltip>
  );
  const editingIcons = (
    <Stack direction="row">
      <Tooltip title="Save" placement="top">
        <IconButton
          disableRipple
          size="small"
          type="button"
          sx={{ ml: 1 }}
          onClick={() => {
            onSave(inputData);
            setEditing(false);
          }}
        >
          <Check />
        </IconButton>
      </Tooltip>
      <Tooltip title="Cancel" placement="top">
        <IconButton
          disableRipple
          size="small"
          type="button"
          sx={{ ml: 1 }}
          onClick={() => {
            setInputData(children);
            setEditing(false);
          }}
        >
          <Close />
        </IconButton>
      </Tooltip>
    </Stack>
  );

  return (
    <InputBase
      size="small"
      fullWidth
      placeholder="Name"
      endAdornment={
        editting ? editingIcons : showEditIcon ? editIcon : undefined
      }
      defaultValue={children}
      readOnly={!editting}
      autoFocus={editting}
      onMouseOver={() => {
        setShowEditIcon(showAlways || true);
      }}
      onMouseLeave={() => {
        setShowEditIcon(showAlways || false);
      }}
      onBlur={(e) => {
        setInputData(e.currentTarget.value);
      }}
      sx={{ borderBottom: editting ? 1 : 0 }}
      {...props}
    />
  );
};
