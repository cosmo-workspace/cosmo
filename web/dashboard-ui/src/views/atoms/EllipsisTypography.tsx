import {
  Tooltip,
  TooltipProps,
  Typography,
  TypographyProps,
} from "@mui/material";
import React from "react";

export type EllipsisTypographyProps = Omit<TypographyProps, "children"> & {
  children?: string;
  placement?: TooltipProps["placement"];
};

export const EllipsisTypography: React.FC<EllipsisTypographyProps> = (
  props
) => {
  const [isOverflow, setIsOverflow] = React.useState(false);
  const paragraph = React.useRef<HTMLSpanElement>(null);

  React.useEffect(() => {
    const pElement = paragraph.current;
    if (pElement) {
      setIsOverflow(Boolean(pElement.offsetWidth < pElement.scrollWidth));
    }
  }, [paragraph]);

  const title = props.children || "";

  // return isOverflow ? (<Tooltip title={title} placement={placement} > {typoglaphy} </Tooltip >) : typoglaphy
  return isOverflow ? (
    <Tooltip title={title} placement={props.placement}>
      <Typography
        ref={paragraph}
        variant="caption"
        display="block"
        sx={{
          textOverflow: "ellipsis",
          whiteSpace: "nowrap",
          overflow: "hidden",
        }}
        {...props}
      >
        {props.children}
      </Typography>
    </Tooltip>
  ) : (
    <Typography
      ref={paragraph}
      variant="caption"
      display="block"
      sx={{
        textOverflow: "ellipsis",
        whiteSpace: "nowrap",
        overflow: "hidden",
      }}
      {...props}
    >
      {props.children}
    </Typography>
  );
};
