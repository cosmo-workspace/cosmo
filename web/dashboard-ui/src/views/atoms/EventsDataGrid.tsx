import { Info, Warning } from "@mui/icons-material";
import { Chip, SxProps } from "@mui/material";
import {
  DataGrid,
  DataGridProps,
  GridColDef,
  gridClasses,
} from "@mui/x-data-grid";
import React from "react";
import { Event } from "../../proto/gen/dashboard/v1alpha1/event_pb";
import { EventDetailDialogContext } from "../organisms/EventDetailDialog";

export type EventsDataGridProp = {
  events: Event[];
  maxHeight?: number;
  sx?: SxProps;
  dataGridProps?: Omit<DataGridProps, "columns" | "sx">;
  clock: Date;
};

export const EventsDataGrid: React.FC<EventsDataGridProp> = ({
  events,
  maxHeight,
  sx,
  dataGridProps,
  clock,
}) => {
  const eventDetailDialogDispatch = EventDetailDialogContext.useDispatch();

  const columns: GridColDef[] = [
    { field: "type", headerName: "Type", width: 50 },
    {
      field: "eventTime",
      headerName: "LastSeen",
      valueGetter: (value, row: Event) =>
        (row.series?.lastObservedTime || row.eventTime)?.toDate(),
      valueFormatter: (value: Date) =>
        formatTime(clock.getTime() - value.getTime()),
      flex: 0.2,
      minWidth: 80,
    },
    {
      field: "reason",
      headerName: "Reason",
      flex: 0.25,
      renderCell: (params) => (
        <Chip
          icon={
            params.api.getRow(params.id).type == "Normal" ? (
              <Info color="success" />
            ) : (
              <Warning color="warning" />
            )
          }
          label={params.value}
        />
      ),
    },
    { field: "regardingWorkspace", headerName: "Workspace" },
    {
      field: "regarding",
      headerName: "Object",
      valueGetter: (value, row) =>
        row.regarding?.kind + "/" + row.regarding?.name,
      flex: 0.3,
    },
    { field: "reportingController", headerName: "Reporter" },
    {
      field: "series",
      headerName: "Count",
      valueGetter: (value, row) => row.series?.count || 1,
    },
    { field: "note", headerName: "Message", flex: 1 },
  ];

  return (
    <>
      <div style={{ width: "100%", maxHeight: maxHeight, minHeight: 100 }}>
        <DataGrid
          autoHeight={maxHeight === undefined}
          rows={events}
          columns={columns}
          getRowHeight={() => "auto"}
          sx={{
            [`& .${gridClasses.cell}`]: {
              py: 1,
            },
            ...sx,
          }}
          onRowDoubleClick={(params) => {
            eventDetailDialogDispatch(true, { event: params.row });
          }}
          hideFooter
          {...dataGridProps}
        />
      </div>
    </>
  );
};

export function formatTime(milisec: number): string {
  const sec = Math.floor(milisec / 1000) % 60;
  const min = Math.floor(milisec / 1000 / 60) % 60;
  const hours = Math.floor(milisec / 1000 / 60 / 60) % 24;
  if (hours > 0) return `${hours}h${min}m${sec}s`;
  if (min > 0) return `${min}m${sec}s`;
  return `${sec}s`;
}
