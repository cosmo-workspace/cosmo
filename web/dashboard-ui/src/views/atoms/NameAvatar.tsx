import { Avatar, AvatarProps, Typography } from "@mui/material";
import { AccountCircle } from "@mui/icons-material";
import React from "react";

export const NameAvatar: React.VFC<{ name?: string } & AvatarProps> = (props) => {
  return (
    props.name ?
      <Avatar {...props} sx={{ bgcolor: stringToColor(props.name), ...props.sx }}>
        <Typography sx={{ color: (theme) => (theme.palette.mode === 'light' ? 'white' : 'black') }} fontSize='inherit'>
          {props.name.substring(0, 1).toUpperCase()}
        </Typography>
      </Avatar>
      :
      <Avatar {...props} ><AccountCircle /></Avatar>
  )
}

const stringToColor = (str: string) => {
  let hash = 0;
  for (let i = 0; i < str.length; i++) {
    hash = str.charCodeAt(i) + ((hash << 5) - hash);
  }
  let color = '#';
  for (let i = 0; i < 3; i++) {
    let value = (hash >> (i * 8)) & 0xFF;
    color += ('00' + value.toString(16)).substr(-2);
  }
  return color;
}
