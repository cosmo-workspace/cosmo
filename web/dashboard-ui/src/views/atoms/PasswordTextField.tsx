import {
  IconButton,
  InputAdornment, TextField, TextFieldProps
} from "@mui/material";
import { Visibility, VisibilityOff, VpnKey } from "@mui/icons-material";
import React, { useState } from "react";


export const PasswordTextField: React.VFC<TextFieldProps> = ({ type, ...props }) => {
  const [isPassShow, setIsPassShow] = useState(false);

  return (
    <TextField {...props} type={isPassShow ? 'text' : 'password'}
      InputProps={{
        startAdornment: (<InputAdornment position="start"><VpnKey /></InputAdornment>),
        endAdornment: (<InputAdornment position="end">
          <IconButton
            tabIndex={-1}
            onPointerDown={() => { setIsPassShow(true) }}
            onPointerUp={() => { setIsPassShow(false) }}
            onPointerOut={() => { setIsPassShow(false) }}
          >
            {isPassShow ? <Visibility /> : <VisibilityOff />}
          </IconButton>
        </InputAdornment>),
      }}
    />
  );
}
