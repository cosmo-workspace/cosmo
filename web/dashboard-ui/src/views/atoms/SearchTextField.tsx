import { Clear, SearchTwoTone } from "@mui/icons-material";
import {
  IconButton,
  InputAdornment,
  TextField,
  TextFieldProps,
} from "@mui/material";
import React from "react";

export const SearchTextField: React.FC<
  {
    search: string;
    setSearch: (search: string) => void;
  } & TextFieldProps
> = ({ search, setSearch, ...props }) => {
  return (
    <TextField
      InputProps={
        search !== ""
          ? {
              startAdornment: (
                <InputAdornment position="start">
                  <SearchTwoTone />
                </InputAdornment>
              ),
              endAdornment: (
                <InputAdornment position="end">
                  <IconButton
                    size="small"
                    tabIndex={-1}
                    onClick={() => {
                      setSearch("");
                    }}
                  >
                    <Clear />
                  </IconButton>
                </InputAdornment>
              ),
            }
          : {
              startAdornment: (
                <InputAdornment position="start">
                  <SearchTwoTone />
                </InputAdornment>
              ),
            }
      }
      placeholder="Search"
      size="small"
      value={search}
      onChange={(e) => setSearch(e.target.value)}
      sx={{ flexGrow: 0.5 }}
      {...props}
    />
  );
};
