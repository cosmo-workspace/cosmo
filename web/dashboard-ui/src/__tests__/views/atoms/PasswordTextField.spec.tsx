import { cleanup, fireEvent, render } from '@testing-library/react';
import React from 'react';
import { afterEach, describe, expect, it } from "vitest";
import { PasswordTextField } from '../../../views/atoms/PasswordTextField';

afterEach(cleanup);


describe('PasswordTextField', () => {

  it('PasswordTextField', () => {
    const target = render(
      <PasswordTextField label="Password" margin="normal" fullWidth />,
    );
    const { asFragment, container } = target;

    expect(asFragment()).toMatchSnapshot();
    fireEvent.pointerDown(container.getElementsByClassName('MuiIconButton-root')[0]);
    expect(asFragment()).toMatchSnapshot();
    fireEvent.pointerUp(container.getElementsByClassName('MuiIconButton-root')[0]);
    expect(asFragment()).toMatchSnapshot();

    fireEvent.pointerDown(container.getElementsByClassName('MuiIconButton-root')[0]);
    expect(asFragment()).toMatchSnapshot();
    fireEvent.pointerOut(container.getElementsByClassName('MuiIconButton-root')[0]);
    expect(asFragment()).toMatchSnapshot();
  });
});