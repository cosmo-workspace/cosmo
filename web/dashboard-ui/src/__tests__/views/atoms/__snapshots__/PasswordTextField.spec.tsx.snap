// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordTextField PasswordTextField 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Password
    </label>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1gywuxd-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="VpnKeyIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12.65 10C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H17v4h4v-4h2v-4H12.65zM7 14c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
          />
        </svg>
      </div>
      <input
        aria-invalid="false"
        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
        type="password"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="VisibilityOffIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-186xcr5"
        >
          <span>
            Password
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PasswordTextField PasswordTextField 2`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Password
    </label>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1gywuxd-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="VpnKeyIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12.65 10C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H17v4h4v-4h2v-4H12.65zM7 14c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
          />
        </svg>
      </div>
      <input
        aria-invalid="false"
        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
        type="text"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="VisibilityIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-186xcr5"
        >
          <span>
            Password
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PasswordTextField PasswordTextField 3`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Password
    </label>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1gywuxd-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="VpnKeyIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12.65 10C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H17v4h4v-4h2v-4H12.65zM7 14c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
          />
        </svg>
      </div>
      <input
        aria-invalid="false"
        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
        type="password"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="VisibilityOffIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-186xcr5"
        >
          <span>
            Password
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PasswordTextField PasswordTextField 4`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Password
    </label>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1gywuxd-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="VpnKeyIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12.65 10C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H17v4h4v-4h2v-4H12.65zM7 14c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
          />
        </svg>
      </div>
      <input
        aria-invalid="false"
        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
        type="text"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="VisibilityIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-186xcr5"
        >
          <span>
            Password
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PasswordTextField PasswordTextField 5`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
    >
      Password
    </label>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-1gywuxd-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="VpnKeyIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12.65 10C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H17v4h4v-4h2v-4H12.65zM7 14c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
          />
        </svg>
      </div>
      <input
        aria-invalid="false"
        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-152mnda-MuiInputBase-input-MuiOutlinedInput-input"
        type="password"
        value=""
      />
      <div
        class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="VisibilityOffIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-186xcr5"
        >
          <span>
            Password
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
